CoffeeMachine_Task1/src/bg/coffee_machine/PremiumCoffeeMachine.java

```diff
@@ -0,0 +1,71 @@
+package bg.coffee_machine;
+
+public class PremiumCoffeeMachine implements CoffeeMachine {
+    private PremiumContainer container;
+    private boolean autoRefill = false;
+
+    public PremiumCoffeeMachine(boolean autoRefill) {  this(); this.autoRefill = autoRefill; }
+    public PremiumCoffeeMachine() {
+        this.container = new PremiumContainer();
+    }
+
+    @Override
+    public Product brew(Beverage beverage) {
+        if(beverage instanceof Espresso)
+            if(container.enoughEspresso()){
+                orderEspresso();
+                return new Product(beverage, "Premium");
+            }
+            else if(autoRefill){
+                refill();
+                orderEspresso();
+            }
+        if(beverage instanceof Cappuccino)
+            if(container.enoughCappuccino()) {
+                orderCappuccino();
+                return new Product(beverage, "Premium");
+            }
+            else if(autoRefill){
+                refill();
+                orderCappuccino();
+            }
+        if(beverage instanceof Mochaccino)
+            if(container.enoughMochaccino()){
+                orderMochaccino();
+                return new Product(beverage, "Premium");
+            }
+            else if(autoRefill){
+                refill();
+                orderMochaccino();
+            }
+        return null;
+    }
+
+    @Override
+    public Container getSupplies() {
+        return container;
+    }
+
+    @Override
+    public void refill() {
+        // use const??
+        this.container.setCacao(300);
+        this.container.setCoffee(1000);
+        this.container.setMilk(1000);
+        this.container.setWater(1000);
+    }
+
+    private void orderEspresso() {
```

Comment:
This method name means something different than what it does. The same applies for `orderMochaccino` and `orderCappuccino`. Consider renaming them to something more meaningful like - `decreaseSuppliesFor*`